src/store.rs

```diff
@@ -196,6 +196,69 @@ impl PassStore {
         Ok(current)
     }
 
+    /// Initializes the directory structure for the password store. Fails if the directory exists
+    /// and has files or folders or if no secret key can be found for the specified `gpgid`.
+    ///
+    /// # Examples
+    /// 
+    /// ```
+    /// use std::path::PathBuf;
+    /// use rasslib::store::PassStore;
+    /// 
+    /// let store_path = PathBuf::from("/tmp/.store");
+    /// let mut store = PassStore::from(&store_path).unwrap();
+    /// let gpgid = "pb@opatut.de";
+    /// 
+    /// store.init(gpgid).unwrap();
+    /// ```
+    ///
+    pub fn init(&mut self, gpgid: &str) -> Result<()> {
+        let ctx = gpgme::Context::from_protocol(
+            gpgme::Protocol::OpenPgp).unwrap();
+
+
+        if self.passhome.is_dir() {
+            if let Ok(r) = fs::read_dir(self.passhome.clone()) {
+                if r.count() > 0 {
+                    let s = format!("Directory exists and not empty: {:?} ", self.passhome);
+                    return Err(PassStoreError::Other(s));
+                }
+            }
+        }
+
+        match ctx.find_secret_key(gpgid) {
+            Ok(key) => {
+                if ! key.has_secret() {
+                    let s = format!("Secret key for {:?} is not available, wouldn't be able to decrypt passwords.", key.id().unwrap());
+                    return Err(PassStoreError::Other(s))
+                }
+
+                self.gpgid = String::from(key.fingerprint().unwrap());
+            },
+            Err(_) => {
+                let s = format!("Secret key {} not found.", gpgid);
+                return Err(PassStoreError::Other(s))
+            }
+        }
+
+        let gpgid_fname = String::from(PASS_GPGID_FILE);
+        let gpgid_path = self.passhome.clone().join(PathBuf::from(gpgid_fname));
+
+        if let Err(_) = fs::create_dir_all(&self.passhome.clone()) {
+            let s = format!("Failed to create directory: {:?}", self.passhome);
+            return Err(PassStoreError::Other(s))
+        }
+
+        match write_gpgid_to_file(&gpgid_path, &self.gpgid) {
+            Ok(id) => id,
```

Comment:
just match `Err(_)` and `_` cases, since the `write_gpgid_to_file` doesn't return anything in the Ok case